.github/workflows/upload-docs.yaml

```
@@ -0,0 +1,66 @@
+name: Upload Docs
+
+on:
+  pull_request:
+    branches:
+      - main
+      - ci/evm-docs-to-github-pages
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: 'pages'
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
```

```suggestion
      - name: Setup tools
        uses: ./.github/actions/setup-tools
        with:
          install-nodejs: 'true'
```
Reuse setup action, similar to other workflows in this repo